src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VCFComparator.java

```diff
@@ -0,0 +1,887 @@
+        if (getDrivingVariantsFeatureInputs().size() != 2) {
+            throw new UserException.BadInput("VCFComparator expects exactly two inputs -- one actual and one expected.");
+        }
+        final List<FeatureInput<VariantContext>> expected = getDrivingVariantsFeatureInputs().stream().filter(file -> file.getName().equals("expected")).collect(Collectors.toList());
+        if (expected.size() != 1) {
+            throw new UserException.BadInput("Tool requires exactly one expected input file");
+        }
+
+        annotatorEngine = new VariantAnnotatorEngine(makeVariantAnnotations(), null, Collections.emptyList(), false, false);
+
+        isSingleSample = getSamplesForVariants().size() == 1 ? true : false;  //if either expected or actual has more than one sample then we're not single-sample
+
+        if (warningsOutput != null) {
+            outputStream = new PrintStream(warningsOutput.getOutputStream());
+        }
+    }
+
+    @Override
+    public void apply(final List<VariantContext> variantContexts, final ReferenceContext referenceContext, final List<ReadsContext> readsContexts) {
+        if (isSingleSample && !variantContexts.get(0).getAlleles().contains(Allele.NON_REF_ALLELE)) {
+            throw new UserException.BadInput("Single-sample mode expects two GVCFs with <NON_REF> data for comparison");
+        }
+
+        if (variantContexts.size() == 1) {
+            final VariantContext vc = variantContexts.get(0);
+            //TODO: if there's a variant that overlaps the requested interval the start may not be in the requested intervals
+            if (!muteDiffs) {
+                throwOrWarn(new UserException("Unmatched variant in " + vc.getSource() + " at position " + vc.getContig() + ":" + vc.getStart()));
+            //low coverage sites will have QUAL for singleton hom-vars boosted because of GQ0 hom-refs
+            } else if (hasGoodEvidence(vc)) {
+                throwOrWarn(new UserException("Unmatched variant in " + vc.getSource() + " at position " + vc.getContig() + ":" + vc.getStart()));
+            }
+        }
+
+        //here we have all variants overlapping a given position
+        //if there's no actual for the expected (at the same position), see if the expected has an overlapping deletion
+        for (final VariantContext vc : variantContexts) {
+            if (vc.getSource().equals("actual")) {  //there may be more expected than actual variants, but only compare once
+                continue;  //TODO: this means we don't report cases where actual has multiple reference blocks spanning expected as unmatched
+            }
+            //vc is guaranteed to be from expected
+
+            final List<VariantContext> matches = variantContexts.stream().filter(v -> v.getStart() == vc.getStart()).collect(Collectors.toList());
+            //matches includes vc
+            //TODO: matches.size() could be 4 or more, depending on ploidy
+            if (matches.size() == 1 ) {
```

Comment:
We might could use some more comments.  1 match is just expected -- no overlapping variants in expected, no variants with same start in actual.